rules/DowngradePhp85/Rector/Class_/DowngradeFinalPropertyPromotionRector.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp85\Rector\Class_;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Modifiers;
+use PhpParser\Node;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Trait_;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory;
+use Rector\BetterPhpDocParser\PhpDocManipulator\PhpDocTypeChanger;
+use Rector\PhpParser\Node\BetterNodeFinder;
+use Rector\PhpParser\Printer\BetterStandardPrinter;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\MethodName;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/final_promotion
+ *
+ * @see \Rector\Tests\DowngradePhp85\Rector\Class_\DowngradeFinalPropertyPromotionRector\DowngradeFinalPropertyPromotionRectorTest
+ */
+final class DowngradeFinalPropertyPromotionRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Change constructor final property promotion to @final annotation assign', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function __construct(
+        final public string $id
+    ){}
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function __construct(
+        /** @final */
+        public string $id
+    ) {}
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class, Trait_::class];
+    }
+
+    /**
+     * @param Class_|Trait_ $node
+     */
+    public function refactor(Node $node): null
+    {
+        $constructorClassMethod = $node->getMethod(MethodName::CONSTRUCT);
+        if (! $constructorClassMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        foreach ($constructorClassMethod->params as $promotedParam) {
+            if (($promotedParam->flags & Modifiers::FINAL) !== 0) {
+                $promotedParam->flags &= ~Modifiers::FINAL;
```

Comment:
use VisibilityManipulator->makeNonFinal() usage, see 

https://github.com/rectorphp/rector-src/blob/c32b6305df1bf1a01f1a7e3d48dea5316619d505/rules/Privatization/NodeManipulator/VisibilityManipulator.php#L69

Reply:
Ok

Reply:
The `makeNonFinal` method is accepting `Class_|ClassMethod`, not accepting `param`

Reply:
It should add Param as well on rector-src then